File: src/main/java/net/pterodactylus/sone/database/FriendStore.java
package net.pterodactylus.sone.database;

import net.pterodactylus.sone.data.Sone;

/**
 * Stores information about the {@link Sone#getFriends() friends} of a {@link Sone}.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
public interface FriendStore {

	void addFriend(Sone localSone, String friendSoneId);
	void removeFriend(Sone localSone, String friendSoneId);

}

File: src/main/kotlin/net/pterodactylus/sone/database/FriendStore.kt
package net.pterodactylus.sone.database

import net.pterodactylus.sone.data.Sone

/**
 * Stores information about the [friends][Sone.getFriends] of a [Sone].
 */
interface FriendStore {

	fun addFriend(localSone: Sone, friendSoneId: String)
	fun removeFriend(localSone: Sone, friendSoneId: String)

}
